### PR TITLE
queue multiple fcs 

### PR DESCRIPTION
**Pull request by mejango**

### CODE REVIEW

File: contracts/JBFundingCycleStore.sol [hunk 1]
```diff
@@ -465,9 +505,19 @@ contract JBFundingCycleStore is JBControllerUtility, IJBFundingCycleStore {
     // Get the necessary properties for the base funding cycle.
     JBFundingCycle memory _baseFundingCycle = _getStructFor(_projectId, _fundingCycle.basedOn);
 
+    // Find the base cycle that is not still queued.
+    while (_baseFundingCycle.start > block.timestamp) {
```
Comment (xBA5ED) [inline contracts/JBFundingCycleStore.sol:509]: do-while loop?

File: contracts/JBFundingCycleStore.sol [hunk 2]
```diff
@@ -476,35 +526,30 @@ contract JBFundingCycleStore is JBControllerUtility, IJBFundingCycleStore {
...
+    JBFundingCycle memory _fundingCycle = _getStructFor(_projectId, _configuration);
 
-    // If the latest is expired, return an empty funding cycle.
-    // A duration of 0 cannot be expired.
-    if (
-      _fundingCycle.duration > 0 && block.timestamp >= _fundingCycle.start + _fundingCycle.duration
-    ) return 0;
+    // Loop through all most recently configured funding cycles until an eligible one is found, or we've proven one can't exit.
+    while (_fundingCycle.number != 0) {
+      // If the latest is expired, return an empty funding cycle.
```
Comment (xBA5ED) [inline contracts/JBFundingCycleStore.sol:539]: do-while loop?

File: contracts/JBFundingCycleStore.sol [hunk 3]
```diff
@@ -171,17 +194,20 @@ contract JBFundingCycleStore is JBControllerUtility, IJBFundingCycleStore {
       // Get the funding cycle for the latest ID.
       _fundingCycle = _getStructFor(_projectId, _fundingCycleConfiguration);
 
-      // If it's not approved or if it hasn't yet started, get a reference to the funding cycle that the latest is based on, which has the latest approved configuration.
-      if (!_isApproved(_projectId, _fundingCycle) || block.timestamp < _fundingCycle.start)
+      // Get a reference to the ballot state.
+      JBBallotState _ballotState = _ballotStateOf(_projectId, _fundingCycle);
+
+      // While the cycle has a ballot that isn't approved or if it hasn't yet started, get a reference to the funding cycle that the latest is based on, which has the latest approved configuration.
+      while (
```
Comment (xBA5ED) [inline contracts/JBFundingCycleStore.sol:201]: do-while loop?

File: contracts/JBFundingCycleStore.sol [hunk 4]
```diff
@@ -465,9 +505,19 @@ contract JBFundingCycleStore is JBControllerUtility, IJBFundingCycleStore {
     // Get the necessary properties for the base funding cycle.
     JBFundingCycle memory _baseFundingCycle = _getStructFor(_projectId, _fundingCycle.basedOn);
 
+    // Find the base cycle that is not still queued.
+    while (_baseFundingCycle.start > block.timestamp) {
+      // Set the configuration.
+      configuration = _baseFundingCycle.configuration;
+      // Get the funding cycle for the configuration.
+      _fundingCycle = _getStructFor(_projectId, configuration);
```
Comment (xBA5ED) [inline contracts/JBFundingCycleStore.sol:513]: `configuration` and `_fundingCycle` aren't needed in this loop, we can get these after we find the correct `_baseFundingCycle`.

Comment (xBA5ED) [reply contracts/JBFundingCycleStore.sol:513]: Isn't this the same as `_fundingCycle = _baseFundingCycle;`?

File: contracts/JBFundingCycleStore.sol (hunk 3 above)
Comment (mejango) [reply contracts/JBFundingCycleStore.sol:201]: doesnt work here

File: contracts/JBFundingCycleStore.sol (hunk 2 above)
Comment (mejango) [reply contracts/JBFundingCycleStore.sol:539]: works here. done.

File: contracts/JBFundingCycleStore.sol (hunk 1 above)
Comment (mejango) [reply contracts/JBFundingCycleStore.sol:509]: need a while (true) loop with a conditional here. i made the change.

**Pull request closed without merging**